Sources/Datadog/RUM/Instrumentation/Views/RUMViewsHandler.swift

```diff
@@ -193,3 +179,56 @@ internal final class SwiftUIRUMViewsHandler: SwiftUIViewHandler {
         }
     }
 }
+
+extension RUMViewsHandler: UIViewControllerHandler {
+    func notify_viewDidAppear(viewController: UIViewController, animated: Bool) {
+        if let view = stack.first(where: { $0.identity.equals(viewController) }) {
+            // If the stack already contains the view controller, just restarts the view.
+            // This prevent from calling the predicate when unnecessary.
+            add(view: view)
+        } else if let rumView = predicate?.rumView(for: viewController) {
+            add(
+                view: .init(
+                    identity: viewController.asRUMViewIdentity(),
+                    name: rumView.name,
+                    path: rumView.path,
+                    attributes: rumView.attributes
+                )
+            )
+        }
+    }
+
+    func notify_viewDidDisappear(viewController: UIViewController, animated: Bool) {
+        guard stack.contains(where: { $0.identity.equals(viewController) }) else {
+            return
+        }
+
+        remove(identity: viewController.asRUMViewIdentity())
```

Comment:
`remove(identity:)` already does the `contains` check implicitly

Reply:
Ah yes. good catch!